core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -131,6 +155,24 @@ public Integer schemaId() {
     return schemaId;
   }
 
+  private void cacheManifests(FileIO fileIO) {
+    if (fileIO == null) {
+      throw new IllegalStateException("Cannot cache changes: FileIO is null");
```

Comment:
I think this is now an `IllegalArgumentException`.

Reply:
The existing `cacheManifest` function uses `IllegalStateException`, but I agree that `IllegalArgumentException` makes more sense here as we're not relying on state within the class but arguments.